### PR TITLE
New logic

### PR DESCRIPTION
**Pull request by janisBerz**
New logic to handle environments.
There is now a step that will validate if the targeted environmwent is valida and will stop the build from a feature branch when targeting a prod-like environment

### CODE REVIEW

File: jobs/aadAppRegistration.yml [hunk 1]
```diff
@@ -1,38 +1,40 @@
 
 parameters:
-  # build agent pool can be overwritten on build template level 
-  pool: 'Azure-VSTS-VS2017'
-  azureSubscription: 'Reform-CFT-VH-Dev'
```
Comment (ddikman) [inline jobs/aadAppRegistration.yml]: I notice there's a `productionAzureSubscription` below, you don't need this default one?

File: jobs/aadAppRegistration.yml [hunk 2]
```diff
@@ -1,38 +1,40 @@
...
 
   - task: GitVersion@4
     displayName: GitVersion
     inputs:
       preferBundledVersion: false
 
+# Validates if deployment branch is allowed to target a environmet
   - powershell: | 
+      # Check if the input environment name is one of the allowe d environmants.
       if ('${{ parameters.allowedEnvironments }}' -match '$(environmentName)') {Write-Host "Variable validation successful!"}
```
Comment (ddikman) [inline jobs/aadAppRegistration.yml]: new line for contents of if?

Comment (ddikman) [reply jobs/aadAppRegistration.yml]: actually, I noticed the end brackets on most of the ifs are placed a curiously

File: jobs/aadAppRegistration.yml [hunk 3]
```diff
@@ -92,10 +112,11 @@ jobs:
       -AADAppReplyUrls "${{ parameters.aadAppReplyUrls }}"
     displayName: 'Set AAD App Replay URLs'
 
+# Created new DNS records for websites
   - task: AzurePowerShell@3
     displayName: 'Set DNS records for web app'
     inputs:
-      azureSubscription: 'DCD-CFT-VH-Pilot'
+      azureSubscription: ${{ parameters.productionAzureSubscription }}
```
Comment (ddikman) [inline jobs/aadAppRegistration.yml]: i guess this is only for production, or can this parameter contain both production and non-production values?

Comment (janisBerz) [reply jobs/aadAppRegistration.yml]: This is for making DNS changes. The DNS Zone is only in prod subscription.

**Pull request merged**